.github/workflows/docs.yml

```diff
@@ -0,0 +1,103 @@
+name: Documentation
+
+on:
+  workflow_dispatch:
+  pull_requests:
+  push:
+    branches:
+      - main
+      - develop
+  schedule:
+    # Run every day at 3 am UTC
+    - cron: "0 3 * * *"
+
+env:
+  PYBAMM_DISABLE_TELEMETRY: "true"
+  FORCE_COLOR: 3
+  PYBAMM_IDAKLU_EXPR_CASADI: ON
+  PYBAMM_IDAKLU_EXPR_IREE: ON
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout repository code
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      # Install Linux system dependencies
+      - name: Install Linux system dependencies
+        uses: awalsh128/cache-apt-pkgs-action@a6c3917cc929dd0345bfb2d3feaf9101823370ad # v1.4.2
+        with:
+          packages: graphviz pandoc
+          execute_install_scripts: true
+
+      # Install TeXLive for Linux
+      - name: Install TeXLive for Linux
+        run: |
+          sudo apt-get update
+          sudo apt-get install texlive-latex-extra dvipng
+
+      # Set up Python
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          cache: 'pip'
+
+      # Upgrade pip and install dependencies for PyBaMM
+      - name: Install PyBaMM with [docs,all]
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e .[docs,all]
+
+      # Run doctests for Python 3.11
+      - name: Run doctests
+        run: python -m nox -s doctests
+
+      # Build the docs (HTML)
+      - name: Build docs (HTML)
+        run: sphinx-build -b html docs/ docs/_build/html
+
+      # Check links in the documentation with Sphinx
+      - name: Run Sphinx linkcheck
+        run: sphinx-build -b linkcheck docs/ docs/_build/linkcheck
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # Run Lychee URL checker in parallel (No redundant checkout)
+      - name: Lychee URL checker
```

Comment:
@RohitP2005 I think there was a misunderstanding here. Lychee should stay in its own check. The reason is that it often fails due to connectivity issues.

Splitting gives us the following:
1. If Lychee fails and no URLs were added a developer can safely assume they did not break anything. The failure was likely networking. We cannot make an unstable check required
2. Documentation can break due to updated doc strings, the doctests (testing code blocks in the documentation), or bad links inside the documentation. If this fails it must be checked and we can easily force that to be a required check

I would say that the `nox -s doctests` job should also be put in the "test_docs_build.yml" workflow